### PR TITLE
Update FWCore/ParameterSet headers

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

- Moved ParameterSet headers to src if not needed for API
- replaced getFixedSizedArray.h with generic mechanism to get arbitrary types via call to ParameterSet::get(Untracked)Parameter

#### PR validation:

Code compiles.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: FWCore/ParameterSet/interface/ParameterSet.h [hunk 1]
```diff
@@ -748,7 +787,19 @@ namespace edm {
...
 
   template <>
-  std::vector<InputTag> ParameterSet::getParameter<std::vector<InputTag> >(char const* name) const;
+  std::vector<InputTag> ParameterSet::getParameter<std::vector<InputTag>>(char const* name) const;
+
+  // ----------------------------------------------------------------------
+  // ESInputTag
+
+  template <>
+  ESInputTag ParameterSet::getParameter<ESInputTag>(char const* name) const;
```
Comment (makortel) [inline FWCore/ParameterSet/interface/ParameterSet.h:796]: This is beyond this PR, but I want to wonder out loud the usefulness of the `(char const* name)` overloads given that under the hood a `std::string` gets constructed anyway
https://github.com/cms-sw/cmssw/blob/53c50cdd24c436cca4892e8fb7e0d5b5b59b641e/FWCore/ParameterSet/src/ParameterSet.cc#L219

Should we (later) look into replacing both `std::string const&` and `char const*` overloads with a single `std::string_view`?

**Comment by makortel** [comment]
+1

File: FWCore/ParameterSet/interface/ParameterSet.h (hunk 1 above)
Comment (Dr15Jones) [reply FWCore/ParameterSet/interface/ParameterSet.h:796]: I would like to explore that. I think one reason for both `std::string` and `char const*` is how templated functions fall within the overload function set. I'm guessing in the past we needed both else automatic conversion was not happening.

Comment (wddgit) [reply FWCore/ParameterSet/interface/ParameterSet.h:796]: It has been a long time and I might not be remembering this correctly, but I believe the reason we added both interfaces was that use of the char const* interface was causing the conversion to string to be inlined in many many places during the build, which was not efficient. I am remembering what other people said. This is not something I tested myself.

**Comment by qliphy** [comment]
+1

**Pull request merged**